frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -101,11 +105,23 @@ function BookInfo:show(file, book_props)
         end
         table.insert(kv_pairs, { prop_text, prop })
     end
+    -- cover image
     local is_doc = self.document and true or false
-    local viewCoverImage = function()
-        self:onShowBookCover(file)
+    if self.custom_book_cover == nil then
+        self.custom_book_cover = DocSettings:findCoverFile(file)
+    end
+    local callback = function()
+        self:onShowBookCover(file, true)
+    end
+    table.insert(kv_pairs, { _("Cover image:"), _("Tap to display"), callback=callback,
+        separator=is_doc and not self.custom_book_cover })
+    -- custom cover image
+    if self.custom_book_cover then
+        callback = function()
```

Comment:
Just wondering: we assign a function to `callback`, and provide callback to the kv_pairs item.
When then assign another function to that same `callback`, and provide callback to the other kv_pairs item.
Sure that 2nd function doesn't end up getting the one called by the first kv_pairs item?
(Might be fine when passed as function parameters, but not if it were an upvalue/closure - just want to be sure.)

Might be as weel fine as just inlining it in each table.insert(...)

Reply:
Inline doesn't work.
The same name works fine. Have changed to different names for better readability.